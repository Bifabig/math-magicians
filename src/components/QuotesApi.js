import { useEffect, useState } from 'react';

const QuotesApi = () => {
  const [quotes, setQuotes] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const options = {
          method: 'GET',
          headers: {
            'x-api-key': 'btAEiRrit9u1L1sZDM0Oew==si2rJMo7DAvCioHN',
            'Content-Type': 'application/json',
          },
        };
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=happiness',
          options,
        );
        const json = await res.json();
        setQuotes(json);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setQuotes, setIsLoading]);

  return { quotes, hasError, isLoading };
};

export default QuotesApi;
