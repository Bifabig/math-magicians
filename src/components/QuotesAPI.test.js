import * as API from './QuotesApi';

test('api fetch works', async () => {
  const data = await API.quotes;
  expect(data).toMatchSnapshot();
});
