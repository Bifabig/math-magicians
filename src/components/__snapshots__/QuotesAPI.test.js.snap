// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api fetch works 1`] = `undefined`;
